posthog/temporal/batch_exports/temporary_file.py

```diff
@@ -0,0 +1,523 @@
+        )
+        self.field_names = field_names
+        self.extras_action: typing.Literal["raise", "ignore"] = extras_action
+        self.delimiter = delimiter
+        self.quote_char = quote_char
+        self.escape_char = escape_char
+        self.line_terminator = line_terminator
+        self.quoting = quoting
+
+        self._csv_writer: csv.DictWriter | None = None
+
+    @property
+    def csv_writer(self) -> csv.DictWriter:
+        if self._csv_writer is None:
+            self._csv_writer = csv.DictWriter(
+                self.batch_export_file,
+                fieldnames=self.field_names,
+                extrasaction=self.extras_action,
+                delimiter=self.delimiter,
+                quotechar=self.quote_char,
+                escapechar=self.escape_char,
+                quoting=self.quoting,
+                lineterminator=self.line_terminator,
+            )
+
+        return self._csv_writer
+
+    def _write_record_batch(self, record_batch: pa.RecordBatch) -> None:
+        """Write records to a temporary file as CSV."""
+        self.csv_writer.writerows(record_batch.to_pylist())
+
+
+class ParquetBatchExportWriter(BatchExportWriter):
+    """A `BatchExportWriter` for Apache Parquet format.
+
+    We utilize and wrap a `pyarrow.parquet.ParquetWriter` to do the actual writing. We default to their
+    defaults for most parameters; however this class could be extended with more attributes to pass along
+    to `pyarrow.parquet.ParquetWriter`.
+
+    See the pyarrow docs for more details on what parameters can the writer be configured with:
+    https://arrow.apache.org/docs/python/generated/pyarrow.parquet.ParquetWriter.html
+
+    In contrast to other writers, instead of us handling compression we let `pyarrow.parquet.ParquetWriter`
+    handle it, so `BatchExportTemporaryFile` is always initialized with `compression=None`.
+
+    Attributes:
+        schema: The schema used by the Parquet file. Should match the schema of written RecordBatches.
+        compression: Compression codec passed to underlying `pyarrow.parquet.ParquetWriter`.
+    """
+
+    def __init__(
+        self,
+        max_bytes: int,
+        flush_callable: FlushCallable,
+        schema: pa.Schema,
+        compression: str | None = "snappy",
+    ):
+        super().__init__(
+            max_bytes=max_bytes,
+            flush_callable=flush_callable,
+            file_kwargs={"compression": None},  # ParquetWriter handles compression
+        )
+        self.schema = schema
+        self.compression = compression
+
+        self._parquet_writer: pq.ParquetWriter | None = None
+
+    @property
+    def parquet_writer(self) -> pq.ParquetWriter:
+        if self._parquet_writer is None:
+            self._parquet_writer = pq.ParquetWriter(
+                self.batch_export_file,
+                schema=self.schema,
+                # Compression *can* be `None`.
+                compression=self.compression,
```

Comment:
Comment here because my lsp was marking a type error, although mypy is happy apparently.

Reply:
I gave this another look. Landed in this cdef function that seems to parse our compression: https://github.com/apache/arrow/blob/51817917e1436c8799ed382d160798060cd76652/python/pyarrow/_parquet.pyx#L1373

I think it _can't_ be none, but it's technically supported by arrow skipping that check if the compression is not a string: https://github.com/apache/arrow/blob/51817917e1436c8799ed382d160798060cd76652/python/pyarrow/_parquet.pyx#L1854.

Anyways, I've switched to using "none" instead and seems like `"none"` compression works just fine :+1: Moved to that to be more in line with arrow docs.